NewForce-Cohort-6/tabloidfullstack-mac-cheese
# Create a comment jj

This is addressing ticket #20 Create A Comment

[Link to Trello Card](https://trello.com/c/NFAouxao)

As a commenter, I would like to be able to add a Comment to a Post so that I can take part in the discussion about a Post.

*Given* the user is viewing the Details of a Post*When* they select the Add Comment menu option*Then* they should be directed to a page with a form for creating a new Comment

*Given* the user has entered the relevant information for a Comment*When* they click the Save button*Then* the Comment should be saved to the database*And* the creation datetime should be automatically set to the current date and time*And* the user should be redirected to the new Comments list page for the related Post

The "relevant information" for a Comment is

* Subject
* Content

<hr>

## Testing Instructions

In your terminal, from your current branch:
* `git add` and `git commit` whatever you're working on
* `git checkout main`
* `git fetch -a` or `git fetch all` to get the branch
* If files do not show, `git pull origin CreateAComment-JJ`
* `git checkout CreateAComment-JJ`
* start the application with `npm start` in your terminal
* Ensure that you have created and are running the Tabloid database

### Data (if needed)

[Fill out the tables and columns](https://github.com/NewForce-Cohort-6/tabloidfullstack-mac-cheese/blob/main/SQL/01_Tabloid_Create_DB.sql)

[Seed data](https://github.com/NewForce-Cohort-6/tabloidfullstack-mac-cheese/blob/main/SQL/02_Tabloid_Seed_Data.sql)

### To Test

* Launch the api from VisualStudio using the filled in green 'play' button at the top.
* Lauch the front end code by right clicking the 'client' folder in visual studio solution explorer and selecting 'Open in terminal'
* Once the terminal opens, type 'code .'
* This should open visual studio code. Open a new terminal and type 'npm start'
* In the browser, if not logged in already, register an email with the registration button, then log in with that email
* Once logged in, click the posts button in the navbar at the top.
* You should see a list of posts, click the title of any post to be taken to a details page
* On the post details page, you should be able to see comments (if any), and a "add comment" button at the bottom.
* Click the add comment button.
* You should be taken to a form page, fill out a comment then hit the "Save" button.
* You should be taken back to the post details page, with your new comment displaying.
